li-protobuf/src/main/java/com/linkedin/data/protobuf/TextBuffer.java

```diff
@@ -0,0 +1,73 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+package com.linkedin.data.protobuf;
+
+/**
+ * A container for holding text data in the form of a char[] to minimize allocations when parsing strings.
+ */
+final class TextBuffer
+{
+  private char[] _buffer;
+
+  /**
+   * Constructor
+   *
+   * @param intialSize  The initial size of the buffer instantiated from the pool.
+   */
+  public TextBuffer(int intialSize)
+  {
+    _buffer = new char[intialSize];
+  }
+
+  /**
+   * Get a buffer of the given size from this instance. If the underlying instance size greater than or equal
+   * to the requested size, the underlying buffer is returned as is. Else, the existing buffer is recycled, and
+   * a new buffer of at least the given size is retrieved from the underlying pool.
+   */
+  public char[] getBuf(int size)
+  {
+    if (_buffer == null)
+    {
+      throw new IllegalStateException("Buffer already in use or closed.");
+    }
+
+    if (_buffer.length >= size)
+    {
+      char[] buffer = _buffer;
+      _buffer = null;
+      return buffer;
```

Comment:
I want to nil the _buffer (ivar) to make sure that no one ends up using the underlying buffer concurrently in case of bugs with the code. Such races may result in String corruption.

Reply:
is `ivar` objective-c concept? lol